plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -2791,10 +2791,19 @@ public function read_meta( &$object ) {
 	 * @return bool
 	 */
 	public function delete_meta( &$object, $meta ) {
+		if ( $this->should_backfill_post_record() && isset( $meta->id ) && ! isset( $meta->key ) ) {
+			// Let's get the actual meta key before its deleted for backfilling. We cannot delete just by ID because meta IDs are different in HPOS and posts tables.
```

Comment:
As you say here, the meta ID here is different from the meta ID from the postmeta table, so perhaps it wouldn't hurt to be explicit in the method docblock to make sure people won't call this with wrong IDs.